src/enclave/enclave.h

```diff
@@ -204,6 +204,9 @@ namespace enclave
           bp, AdminMessage::stop, [&bp, this](const uint8_t*, size_t) {
             bp.set_finished();
             threading::ThreadMessaging::thread_messaging.set_finished();
+            auto w = writer_factory.create_writer_to_outside();
```

Comment:
I think this should go at the end of `run_main` in `enclave.h` - anything after `bp.run()` will only execute once this `set_finished` has been processed, so this moves the notification slightly closer to the point where we actually exit.